spark/src/main/scala/org/apache/spark/sql/comet/CometBroadcastExchangeExec.scala

```diff
@@ -257,9 +257,28 @@ class CometBatchRDD(
   }
 
   override def compute(split: Partition, context: TaskContext): Iterator[ColumnarBatch] = {
-    val partition = split.asInstanceOf[CometBatchPartition]
+    new Iterator[ColumnarBatch] {
+      val partition = split.asInstanceOf[CometBatchPartition]
+      val batchesIter = partition.value.value.map(CometExec.decodeBatches(_)).toIterator
+      var iter: Iterator[ColumnarBatch] = null
+
+      override def hasNext: Boolean = {
+        if (iter != null) {
+          if (iter.hasNext) {
+            return true
+          }
+        }
+        if (batchesIter.hasNext) {
+          iter = batchesIter.next()
```

Comment:
is it possible that this `batchesIter.next` has no more next but the next item in `batchesIter` still has more elements? e.g.: `[1, 2, 3], [], [4, 5, 6]`?

Reply:
I restored this part.